cmd/engine/engine.go

```diff
@@ -178,6 +178,30 @@ func doCmd(_ *cobra.Command, _ []string) (err error) {
 	assessmentService = service_assessment.NewService()
 	evidenceStoreService = service_evidenceStore.NewService()
 
+	assessmentService.RegisterAssessmentResultHook(func(result *assessment.AssessmentResult, err error) {
+
+		if err != nil {
+			log.Errorf("error was passed: %v", err)
+		}
+		_, err = orchestratorService.StoreAssessmentResult(context.Background(), &orchestrator.StoreAssessmentResultRequest{
```

Comment:
So every time the hook function is called, the assessment result is sent to the orchestrator? I would say the assessment result should be always forwarded to the orchestrator as soon it is created in the code of `assessment.go`.

Reply:
But exactly that is happening in service/assessment.go line 149/178 (updated file) 

How am I able to reference to a particular line of code?

Reply:
Then why do we need the `StoreAssessmentResult` in the hook as well? Aren't we sending the assessment result twice in this way?

I also don't know how to reference to specific LOC :)

Reply:
No, the StoreAssessmentResult in the hook is from the orchestrator which is called. 

Perhaps we should have a call about that :)

Reply:
I understand it now. IMO, the assessment should send the assessment results to the orchestrator within it's code via gRPC and not directly calling the orchestrator function within the result hook. I think, we can leave it like that for now. And in the future "workflow PR" we can discuss this further.